src/mpid/ch4/netmod/ofi/ofi_events.c

```diff
@@ -225,7 +228,9 @@ static int recv_huge_event(struct fi_cq_tagged_entry *wc, MPIR_Request * rreq)
     } else {
         recv_elem->cur_offset = MPIDI_OFI_global.max_msg_size;
     }
-    MPIDI_OFI_get_huge_event(NULL, (MPIR_Request *) recv_elem);
+    if (ready_to_get) {
+        MPIDI_OFI_get_huge_event(NULL, (MPIR_Request *) recv_elem);
```

Comment:
This inversion seems unnecessary. Why not just put this line at the end of the original `if (recv_elem == NULL) { }` block? Flipping things around just makes it act like a big `goto`.